content/develop/concepts/architecture/fragments

```diff
@@ -0,0 +1,143 @@
+---
+title: Working withwq fragments and partial reruns
+slug: /develop/concepts/architecture/fragments
+---
+
+# Working with fragments and partial reruns
+
+Reruns are a central part of every Streamlit app. When users interact with widgets, your script reruns from top to bottom and your app's frontend is updated. Streamlit provides several features to help you develop your app within this execution model. Fragments were introduced in Streamlit version 1.33.0 to allow rerunning a portion or your code instead of your full script. As your app grows larger and more complex, partial reruns help your app efficient and performant.
+
+Before you read about fragments, we recommend having a basic understanding of Session State and forms.
+
+## Use cases for fragments
+
+Fragments are versatile and applicable to a wide variety of circumstances. Here are just a few, common scenarios where fragments are useful:
+
+- Your app has multiple visualizations and each one takes time to load, but you have a filter input that only updates one of them.
+- You have a dynamic form that doesn't need to update the rest of your app (until the form is complete).
+- You want to automatically update a single component or group of components to stream data.
+
+## Defining and calling a fragment
+
+Streamlit provides a decorator to turn any function into a fragment function. When you call a fragment function that contains a widget, a user triggers a _partial rerun_ instead of a full rerun when they interact with that fragment's widget. During a partial rerun, your fragment function is re-executed. Anything within the main body of your fragment is updated on the frontend while the rest of your app remains the same. We will discuss what happens when you write to other containers within a fragment later on.
```

Comment:
Do we want to highlight the "partial rerun" phrase? Will that be confusing? (my intuition is NOT to but it's not a strong intuition, honest question)

Reply:
Between partial rerun and fragment, I feel like more people would guess the correct idea from the former vs the latter. So for search especially, I hypothesize that introducing it as a concept may be helpful. I can try to validate this one way or the other by watching search terms and traffic.

Reply:
Sounds good 👍